### PR TITLE
fix: [CDS-35705]: Yaml Validation for environment and service selection

### PR DESCRIPTION
**Pull request by atharvatrivedi1**
…on added

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32696)

### CODE REVIEW

**Comment by atharvatrivedi1** [comment]
trigger messagecheck

**Comment by hingerabhinav** [comment]
*[`400-rest/src/main/java/software/wings/service/impl/yaml/service/YamlServiceImpl.java` line 350 at r1](https://reviewable.io/reviews/harness/harness-core/32696#-N12CqXo6B0ou18UEL9e:-N12CqXo6B0ou18UEL9f:beqv8e7) ([raw file](https://github.com/harness/harness-core/blob/60c2dd2be1a8434d5fa70db6753c8a681b67b7f2/400-rest/src/main/java/software/wings/service/impl/yaml/service/YamlServiceImpl.java#L350)):*
> ```Java
>           throw new WingsException("serviceSelection and envSelection missing in Yaml File");
>         }
>         if (!freezeWindow.has("serviceSelection")) {
> ```

there might be windows without serviceSelection right? We consider those also as ALL service filter, since freezing based on service was added later.

File: 400-rest/src/main/java/software/wings/service/impl/yaml/service/YamlServiceImpl.java [hunk 1]
```diff
@@ -334,8 +336,26 @@ List<Change> filterInvalidFilePaths(List<Change> changeList) {
...
   }
 
+  @SneakyThrows
   @Override
   public RestResponse<B> update(YamlPayload yamlPayload, String accountId, String entityId) {
+    ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    JsonNode json = mapper.readTree(yamlPayload.getYaml());
+    JsonNode timeRangeBasedFreezeConfigs = json.get("timeRangeBasedFreezeConfigs");
+    for (JsonNode timeRangeBasedFreezeConfig : timeRangeBasedFreezeConfigs) {
+      for (JsonNode freezeWindow : timeRangeBasedFreezeConfig.get("appSelections")) {
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/service/impl/yaml/service/YamlServiceImpl.java:346]: 1. this is at wrong place should be in specific handler class
2.  we are going to merge one PR for userGroupSelection too, if it goes first we need to add that check too or else i will take care

**Review by abhinav-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32696)* status: 5 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/yaml/service/YamlServiceImpl.java` line 159 at r2](https://reviewable.io/reviews/harness/harness-core/32696#-N1Cajcm5mSl-tZHIn6A:-N1Cajcm5mSl-tZHIn6B:b-ov24d) ([raw file](https://github.com/harness/harness-core/blob/d05069db8c4853c5c9644ef0da1e1d0e843ce916/400-rest/src/main/java/software/wings/service/impl/yaml/service/YamlServiceImpl.java#L159)):*
> ```Java
> import com.esotericsoftware.yamlbeans.YamlReader;
> import com.fasterxml.jackson.databind.DeserializationFeature;
> import com.fasterxml.jackson.databind.JsonNode;
> ```

remove unrequired changed

**Review by hingerabhinav — APPROVED** [review]
> Reviewed 4 of 44 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32696)* status: 40 file(s) to review. 1 change requests.

**Review by hingerabhinav — APPROVED** [review]
> Reviewed 22 of 44 files at r3, 19 of 19 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32696)* status: :shipit: complete! ✓

**Review by abhinav-harness — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32696)* status: :shipit: complete! ✓

**Pull request merged**